### PR TITLE
Enhanced logging mechanism and added Cloudwatch support

### PR DESCRIPTION
**Pull request by patilsuraj767**
- Shorter log statements
- Strict use of f strings instead of old string formater

Example -
```
{"@timestamp": "2022-05-31T07:35:34.295Z", "@version": 1, "source_host": "fedora", "name": "yuptoo.processor.utils", "args": [], "levelname": "INFO", "levelno": 20, "pathname": "/home/supatil/redhat-projects/yuptoo/yuptoo/processor/utils.py", "filename": "utils.py", "module": "utils", "stack_info": null, "lineno": 47, "funcName": "download_report", "created": 1653982534.2950478, "msecs": 295.0477600097656, "relativeCreated": 41426.09167098999, "thread": 140488742688576, "threadName": "MainThread", "processName": "MainProcess", "process": 302013, "request_id": "204cb80f84eb/cUpsSnAA7m-000004", "account": "sysaccount", "org_id": "3340851", "message": "Downloading Report from http://minio:9000/insights-upload-perma/204cb80f84eb/cUpsSnAA7m-000004?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=BQA2GEXO711FVBVXDWKM%2F20220531%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20220531T073514Z&X-Amz-Expires=86400&X-Amz-SignedHeaders=host&X-Amz-Signature=f3de554bb627ab3b17068bb9454e5d063c9cf20048690833b7bbe7a574f29a3e"}
```

### CODE REVIEW

**Review by r14chandra — APPROVED** [review]
Looks great! 👍🏻

File: yuptoo/lib/config.py [hunk 1]
```diff
@@ -1,16 +1,7 @@
...
-def get_logger(name):
-    logging.basicConfig(
-        level='INFO',
-        format='%(asctime)s - %(levelname)s  - %(funcName)s - %(message)s'
-    )
-    return logging.getLogger(name)
-
-
-LOG = get_logger(__name__)
+LOG = logging.getLogger(__name__)
```
Comment (kgaikwad) [inline yuptoo/lib/config.py:4]: Where it is being used?

**Review by kgaikwad — CHANGES_REQUESTED** [review]

File: yuptoo/lib/config.py (hunk 1 above)
Comment (patilsuraj767) [reply yuptoo/lib/config.py:4]: Not used, I will remove it.

Comment (patilsuraj767) [reply yuptoo/lib/config.py:4]: It is used on line 9

File: yuptoo/lib/logger.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            cw_handler.setFormatter(LogstashFormatterV1())
+            cw_handler.addFilter(ContextualFilter())
+            logging.root.addHandler(cw_handler)
+
+
+class ContextualFilter(logging.Filter):
+    """
+    This filter gets the request_id from the message and adds it to
+    each log record. This way we do not have to explicitly retreive/pass
+    around the request_id for each log message
```
Comment (kgaikwad) [inline yuptoo/lib/logger.py:63]:
```suggestion
    This filter gets the request_id and tenant info from the message and adds to
    each log record. This way, every time it won't require to retrieve these details
    per log message.
```

**Review by kgaikwad — CHANGES_REQUESTED** [review]
Added few minor comments. Could you also check log sentences as period is missing at the end?

File: yuptoo/lib/logger.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+from yuptoo.lib.config import LOG_LEVEL
+from logstash_formatter import LogstashFormatterV1
+
+request_obj = None
+logger = None
+
+threadctx = local()
+
+
+def clowder_config():
```
Comment (kgaikwad) [inline yuptoo/lib/logger.py:15]:
```suggestion

# Returns Cloudwatch config values 
def cloudwatch_config_values():
```

**Review by kgaikwad — APPROVED** [review]
@patilsuraj767 - Added a query regarding cloudwatch config and suggestions related to it.
Noticed it now, please let me know your thoughts.

@r14chandra  - do you have any additional comments?

**Pull request merged**